### PR TITLE
Problem6 Solution

### PR DESCRIPTION
**Pull request by ntkathole**

### CODE REVIEW

**Comment by trishnaguha** [comment]
Thanks for the PR @ntkathole . Mind designing it object oriented way?

**Review by ntkathole — COMMENTED** [review]
Added object oriented approach

File: solutions/prob6.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# https://projecteuler.net/problem=6
+# Sum square difference of first 100 natural numbers
+
+import math
+
+class sumSquareDiff(object):
+	
+	def __init__(self, number):
+        	self.number = number
```
Comment (trishnaguha) [inline solutions/prob6.py]: Could you please follow [PEP8](https://www.python.org/dev/peps/pep-0008/) rules? 
You will need to modify indentation, naming convention of class, variables, add spaces.

**Review by trishnaguha — CHANGES_REQUESTED** [review]

File: solutions/prob6.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+	def square(self, no):
+        	return no*no
+
+	def sum(self):
```
Comment (trishnaguha) [inline solutions/prob6.py]: Can you change the method name?
It is suggested not to use any built-in names in Python.

**Review by trishnaguha — CHANGES_REQUESTED** [review]

File: solutions/prob6.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+		sum1=0
+		N = self.number
+        	for i in range(1,N+1):
+			sum1=sum1+i
```
Comment (trishnaguha) [inline solutions/prob6.py]: You can do `sum1 += i` instead ?
I would prefer to use some other variable instead of `sum1`.

**Review by trishnaguha — CHANGES_REQUESTED** [review]

File: solutions/prob6.py (hunk 1 above)
Comment (trishnaguha) [inline solutions/prob6.py]: Change the argument name `number` to `endrange` or something? `number` is kind of confusing.

**Review by trishnaguha — CHANGES_REQUESTED** [review]

File: solutions/prob6.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+	
+	def __init__(self, number):
+        	self.number = number
+
+	def square(self, no):
+        	return no*no
+
+	def sum(self):
+		sum1=0
+		N = self.number
```
Comment (trishnaguha) [inline solutions/prob6.py]: `endrange` is constant here. You do not need to assign the `endrange` to a variable. 
We should keep memory management in mind when writing code.

**Review by trishnaguha — CHANGES_REQUESTED** [review]

File: solutions/prob6.py [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+   	def sumofsquares(self):
+		sumofsquare=0
+        	N = self.number
+        	for i in range(1,N+1):
+			sumofsquare=sumofsquare+ self.square(i)
+		return sumofsquare
+
+		
+	def diff(self):
+		print(self.square(self.sum())-self.sumofsquares())
```
Comment (trishnaguha) [inline solutions/prob6.py]: Use `return` here. We can print when calling the function with the object.

**Review by trishnaguha — CHANGES_REQUESTED** [review]

File: solutions/prob6.py [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+	def sum(self):
+		sum1=0
+		N = self.number
+        	for i in range(1,N+1):
+			sum1=sum1+i
+		return sum1
+	
+   	def sumofsquares(self):
+		sumofsquare=0
+        	N = self.number
```
Comment (trishnaguha) [inline solutions/prob6.py]: Follow [review](https://github.com/trishnaguha/projecteulersolution/pull/4/files#r108051215) .

**Review by trishnaguha — CHANGES_REQUESTED** [review]

**Review by ntkathole — COMMENTED** [review]
updated changes

File: solutions/prob6.py [hunk 7]
```diff
@@ -0,0 +1,29 @@
+# https://projecteuler.net/problem=6
+# Sum square difference of first 100 natural numbers
+
+import math
```
Comment (trishnaguha) [inline solutions/prob6.py:6]: The code does not follow PEP8 rules yet.

**Review by trishnaguha — CHANGES_REQUESTED** [review]

**Review by ntkathole — COMMENTED** [review]
Thanks for reviews...formatted according to PEP8

**Comment by trishnaguha** [comment]
Merging it :-).

**Pull request merged**